tests/test_deployments/test_app.py

```diff
@@ -271,3 +280,83 @@ def get_missing(messages: List[str]) -> List[str]:
         assert not get_missing(before_stopping), get_logs()
         self.tendermint_node.stop()
         assert not get_missing(after_stopping), get_logs()
+
+
+def mock_get_node_command_kwargs() -> Dict:
+    """Get the node command kwargs"""
+    kwargs = {
+        "bufsize": 1,
+        "universal_newlines": True,
+    }
+
+    # Pipe stdout and stderr even if we're not going to read to provoke the buffer fill
+    kwargs["stdout"] = subprocess.PIPE
+    kwargs["stderr"] = subprocess.STDOUT
+
+    if platform.system() == "Windows":  # pragma: nocover
+        kwargs["creationflags"] = subprocess.CREATE_NEW_PROCESS_GROUP  # type: ignore
+    else:
+        kwargs["preexec_fn"] = os.setsid  # type: ignore
+
+    return kwargs
+
+
+class TestTendermintBufferFailing(BaseTendermintServerTest):
+    """Test Tendermint buffer"""
+
+    perform_monitoring = False  # Setting this flag to False and not monitoring makes the buffer to fill and freeze Tendermint
+    debug_tendermint = True  # This will cause more logging -> faster failure
+
+    @classmethod
+    def setup_class(cls) -> None:
+        """Setup the test."""
+        with mock.patch.object(
+            TendermintParams,
+            "get_node_command_kwargs",
+            return_value=mock_get_node_command_kwargs(),
+        ):
+            super().setup_class()
```

Comment:
We needed to mock on the `setup_class` as a normal mocking was not working. Thanks for the help here @Karrenbelt